appcues/src/main/java/com/appcues/statemachine/StateMachine.kt

```diff
@@ -102,11 +104,25 @@ internal class StateMachine(
                     Failure(sideEffect.error)
                 }
                 is PresentContainerEffect -> {
-                    // kick off UI
-                    sideEffect.experience.stepContainers[sideEffect.containerIndex].presentingTrait.present()
-
-                    // wait on the RenderingStep state to flow in from the UI, return that result
-                    sideEffect.completion.await()
+                    try {
```

Comment:
this is the important piece.

try {} around this sideEffect and if we catch AppcuesTraitException we update state to Idling to stop the auto transitions from one state to another to continue. (not sure if thats the right way to do this)

returns the Failure and also notify the errorFlow which ends up being collected by analytics

let me know if I'm missing anything here. there is a unit test for this part